components/availability/src/Availability.svelte

```diff
@@ -331,19 +338,22 @@
         let availability = AvailabilityStatus.FREE; // default
         if (allCalendars.length) {
           for (const calendar of allCalendars) {
-            let availabilityExistsInSlot = calendar.timeslots.some(
-              (block) => time < block.end_time && block.start_time < endTime,
-            );
+            const slot = {
+              start_time: time,
+              end_time: endTime,
+              available_calendars: [],
+            };
+            const slotAvailability = overlap([...calendar.timeslots], slot);
             if (calendar.availability === AvailabilityStatus.BUSY) {
-              if (!availabilityExistsInSlot) {
+              if (slotAvailability.concurrentEvents < capacity) {
```

Comment:
IMO, I think a defensive approach would be better UX because not all end users will be checking the console for errors or developing this in a local env